design/proposals/namespace_label_authenticator.md

```diff
@@ -0,0 +1,573 @@
+
+## Logs
+[//]: # "If the logs are listed in the feature doc, add a link to that section. If not, list them here."
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **ID** | **Scenario** | **Log message** |
+|--------|--------------|-----------------|
+| 0 | An authenticating host is configured with either both or none of `authn-k8s/namespace` or `authn-k8s/namespace-label` | Role must have exactly one of the following required constraints: {0} |
+| 1 | An authenticating host is configured with `authn-k8s/namespace-label`, but the authenticating namespace does not conform | Kubernetes namespace {0} does not match label-selector {1} |
+| 2 | An authenticating host is configured with `authn-k8s/namespace-label`, but its format does not match `"<key1>=<value1>"` | Invalid namespace label selector {0}: must adhere to format "\<key1>=\<value1>" |
+| 3 | Kubernetes Authenticator begins validating `authn-k8s/namespace-label` restriction | Validating resource restriction on request: 'namespace-label' |
+| 4 | Kubernetes Authenticator successfully validates `authn-k8s/namespace-label` restriction | Validated K8s resource. Type:'namespace-label', Selector:'{0}', Namespace:'{1}' |
+
+## Documentation
+[//]: # "Add notes on what should be documented in this solution. Elaborate on where this should be documented, including GitHub READMEs and/or official documentation."
+
+The new configuration option should be included in the official documentation
+describing application identity and permission granularity for the Kubernetes
+Authenticator:
+- [ ] Enterprise documentation, [here](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Integrations/k8s-ocp/k8s-appid-in-k8s.htm?tocpath=Integrations%7COpenShift%252FKubernetes%7C_____2#KubernetesresourcesobjectsinConjur)
+- [ ] Open Source documentation, [here](https://docs.conjur.org/Latest/en/Content/Integrations/k8s-ocp/k8s-app-identity.htm?tocpath=Integrations%7COpenShift%252FKubernetes%7CSet%20up%20applications%7C_____5#DefineKubernetesresourcesobjectsasConjuridentities)
+
+The new Rancher integration documentation should include this effort, as it
+enables scoping authentication to a Rancher Project. This documentation is not
+live currently.
+
+## Security
+[//]: # "Are there any security issues with your solution? Even if you mentioned them somewhere in the doc it may be convenient for the security architect review to have them centralized here"
+
+| **Security Issue** | **Description** | **Resolution** |
+|--------------------|-----------------|----------------|
+| **Added Permissions** | The Authenticator identity needs a new permission to get any namespace in a cluster. | This is necessary, and an improvement over requiring permission to list all namespaces in a cluster. List permission means the identity can discover unknown resources, but get permission requires knowledge of existing resources. |
```

Comment:
The permission is required only in the namespaces in which Conjur needs to provide service.
Since we currently already provide permissions on a per namespace level (as a guideline we don't recommend  cluster wide permission to Conjur), then this new permission will be on these namespaces only as well.